server/src/test/java/org/opensearch/rest/RestControllerTests.java

```diff
@@ -52,6 +52,7 @@
 import org.opensearch.http.HttpResponse;
 import org.opensearch.http.HttpServerTransport;
 import org.opensearch.http.HttpStats;
+import org.opensearch.identity.IdentityService;
```

Comment:
Could you add new unit test methods in this class to verify authentication is attempted, it can succeed, if there is an authentication failure the request is rejected?

Reply:
Sure let me add some.

Reply:
So for the RestController it does not seem like there an easy way of adding tests for checking this. 

After reviewing the existing code and tests I came up with something along these lines:

```

  public void testRestAuthenticationSucceeds() {
        Map<String, List<String>> restHeaders = new HashMap<>();
        restHeaders.put("Authorization", Collections.singletonList("Basic YWRtaW46YWRtaW4="));
        final FakeRestRequest fakeRestRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY).withMethod(RestRequest.Method.GET)
            .withPath("/favicon.ico")
            .withHeaders(restHeaders)
            .build();
        final AssertingChannel channel = new AssertingChannel(fakeRestRequest, false, RestStatus.OK);
        restController.dispatchRequest(fakeRestRequest, channel, client.threadPool().getThreadContext());
        assertTrue(channel.getSendResponseCalled());
        assertThat(channel.getRestResponse().contentType(), containsString("image/x-icon"));
    }

    public void testRestAuthenticationFails() {
        Map<String, List<String>> restHeaders = new HashMap<>();
        restHeaders.put("Authorization", Collections.singletonList("Basic sdfjalksdjf;a"));
        final FakeRestRequest fakeRestRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY).withMethod(RestRequest.Method.GET)
            .withPath("/favicon.ico")
            .withHeaders(restHeaders)
            .build();
        final AssertingChannel channel = new AssertingChannel(fakeRestRequest, false, RestStatus.BAD_REQUEST);
        restController.dispatchRequest(fakeRestRequest, channel, client.threadPool().getThreadContext());
        assertTrue(channel.getSendResponseCalled());
        assertFalse(channel.getRestResponse().contentType().contains("image/x-icon"));
    }
    
```

Unfortunately, both tests result in a 200. From what I have determined, there is not any built-in authentication process happening at the RestController level. This makes sense considering the plugin architecture and the way that Security is used as a wrapper for Rest Requests. I am looking into if there are any analogous tests in the Security plugin that may help with setting up what you are asking but I won't have these added by EOD.

Reply:
Want to make an issue for this to follow up later?

Reply:
Yeah I will do that now